### PR TITLE
catch exceptions per addon for a repo check

### PR DESCRIPTION
**Pull request by mzfr**
I have added multiple error as just using bare except was too vague and possibly we can expect multiple exceptions from our checks.

### CODE REVIEW

File: kodi_addon_checker/check_repo.py [hunk 1]
```diff
@@ -12,7 +12,9 @@ def check_repo(repo_path, branch_name, all_repo_addons, pr, config):
...
         if addon_folder[0] != '.':
             addon_path = os.path.join(repo_path, addon_folder)
-            addon_report = check_addon.start(addon_path, branch_name, all_repo_addons, pr, config)
-            repo_report.add(addon_report)
-
+            try:
+                addon_report = check_addon.start(addon_path, branch_name, all_repo_addons, pr, config)
+                repo_report.add(addon_report)
+            except (IOError, SyntaxError, RuntimeError, TypeError):
+                repo_report.add(Record(PROBLEM, "There is some issue with your addon"))
```
Comment (Rechi) [inline kodi_addon_checker/check_repo.py]: @Razzeee do we want to blame the addon?
can the backtrace be reported somewhere?

Comment (mzfr) [reply kodi_addon_checker/check_repo.py]: If not the addon then maybe something general like `Something went wrong` or something along that line.

Comment (Rechi) [reply kodi_addon_checker/check_repo.py]: yes, as long as it counted as error/problem
why not catch all exceptions?

Comment (mzfr) [reply kodi_addon_checker/check_repo.py]: By all exception you mean bare except i.e `except exception as e` ?

**Pull request merged**